### ISSUE TITLE
Home Assistant no entity detected for containers 

### ISSUE THREAD

**Issue by Pl1997** [getwud/wud#97]
Hello,

Thank you for your work, this tool is awesome ! I encountered an issue configuring home-assistant : only one entity is detected, which might be the wud watcher itself : binary_sensor.wud_container_cs_wud . I did enable MQTT and HASS with ENV variables WUD_TRIGGER_MQTT_MOSQUITTO_HASS_ENABLED and WUD_TRIGGER_MQTT_MOSQUITTO_URL.

Here is the log fro WUD (except lines related to cron jobs and watchers, but maybe know that I'm watching 5 containers with their digest, including one that has an update) :

```
01:49:54.036  INFO whats-up-docker/trigger.mqtt.mosquitto: Add hass device [id=wud_container_cs_WUD]
01:49:48.766  INFO whats-up-docker/api: API/UI exposed on port 3000
01:49:49.686  INFO whats-up-docker/watcher.docker.cs: Cron started (0 * * * *)
01:49:49.713  INFO whats-up-docker/watcher.docker.cs: Listening to docker events
01:49:49.796  INFO whats-up-docker/trigger.mqtt.mosquitto: Remove hass device [id=wud_container_cs_WUD]
01:49:45.946  INFO whats-up-docker/trigger.mqtt.mosquitto: Register trigger with configuration {"url":"mqtt://192.168.3.134:1883","hass":{"enabled":true,"prefix":"homeassistant"},"topic":"wud/container","threshold":"all","mode":"single","once":true,"simpletitle":"New ${kind} found for container ${name}","simplebody":"Container ${name} running with ${kind} ${local} can be updated to ${kind} ${remote}\n${link}","batchtitle":"${count} updates available"}
01:49:45.705  INFO whats-up-docker/registry: Register all components of kind trigger for provider mqtt
01:49:45.478  INFO whats-up-docker/prometheus: Init Prometheus module
01:49:45.450  INFO whats-up-docker/store: Load store from (/store/wud.json)
01:49:45.447  INFO whats-up-docker: What's up Docker? is starting (version = latest)
```
And a few lines of the mosquitto addon log in HA : 
```
1642296062: New connection from 192.168.5.170 on port 1883.
1642296062: New client connected from 192.168.5.170 as mqttjs_dd56a9c5 (p2, c1, k60).
1642296160: Saving in-memory database to /data/mosquitto.db.
1642297521: Socket error on client mqttjs_dd56a9c5, disconnecting.
1642297529: New connection from 192.168.5.170 on port 1883.
1642297529: New client connected from 192.168.5.170 as mqttjs_1ed6a2f1 (p2, c1, k60).
1642297778: Socket error on client mqttjs_1ed6a2f1, disconnecting.
1642297786: New connection from 192.168.5.170 on port 1883.
1642297786: New client connected from 192.168.5.170 as mqttjs_50af7f15 (p2, c1, k60).
```

Thank you for your help !

**Comment by fmartinou** [comment]
Hi,

Thank you for your support 😃 .

Are the mosquitto logs `New client connected from 192.168.5.170 ... Socket error on client mqttjs_dd56a9c5, disconnecting.` related to wud connections to mosquitto?

If so, wouldn't there a configuration issue? (authentication, tls...)?
I can see on the wud log that the mqtt trigger is configured as:
```json
{
  "url":"mqtt://192.168.3.134:1883",
  "hass":  {
    "enabled":true,
    "prefix":"homeassistant"
  },
  "topic":"wud/container",
  "threshold":"all",
  "mode":"single",
  "once":true,
  "simpletitle":"New ${kind} found for container ${name}",
  "simplebody":"Container ${name} running with ${kind} ${local} can be updated to ${kind} ${remote}\n${link}","batchtitle":"${count} updates available"
}
```
Is that correct?

**Comment by Pl1997** [comment]
Yes this is correct ! Actually I just "solved" the issue by changing the label wud.tag.include to ^\d+\.\d+\.\d+$$ (it was on ^latest$ ) on a few containers, which instantly showed up in Home Assistant ! So it seems that when tracking the "latest" tag, the integration is not working. I'm not sure whether it a bug or some attended behavior ?

**Comment by fmartinou** [comment]
I must check but it's possible that entities are only created on HA after a first change has been detected.
So it'd explain why your 4 other containers didn't show up 🤔 .

> So it seems that when tracking the "latest" tag, the integration is not working

It should not make a difference that the update is digest related or tag related.
Both should be working.


Regarding the `wud.tag.include` label, be careful to set valid values matching with the tag you 're running and the updates you're expecting.

If you might provide your configuration (wud + container tags + container labels), I can take a look and tell you if I notice something weird.

**Comment by Pl1997** [comment]
> I must check but it's possible that entities are only created on HA after a first change has been detected.

Actually I had thought of that and precisely waited for my first update notification before posting... When a digest update was detected for the first time, nothing new appeared in HA.



> Regarding the `wud.tag.include` label, be careful to set valid values matching with the tag you 're running and the updates you're expecting.

Thank you for the heads up !


> If you might provide your configuration (wud + container tags + container labels), I can take a look and tell you if I notice something weird.

About my configuration, everything is default, except from :

WUD : 
WUD_TRIGGER_MQTT_MOSQUITTO_HASS_ENABLED	true
WUD_TRIGGER_MQTT_MOSQUITTO_URL                            mqtt://192.168.3.134:1883


Invisible containers (tags) :
wud.tag.include        ^latest$
wud.watch.digest.     true

**Comment by alexdelprete** [comment]
I had this issue when I switched from Mosquitto to EMQX as my main homelab MQTT broker.

In EMQX I could see all 60 containers (on 3 docker instances) under the topic `wud/container`, but I didn't see all containers under the `homeassistant/binary_sensor` topic.

I waited a couple of days for the update checks, thinking it would have updated them but that was not the case.

So I finally solved it by stopping WUD container, deleting the wud.json file and recreating the WUD container.

I don't know what's the exact problem, but there was something in wud.json that didn't allow the new broker to receive the necessary updates.

**Comment by alexdelprete** [comment]
@fmartinou every couple of days, unfortunately the WUD entities in HA are either unavailable or not synced to the real status. The only workaround I found was deleting wud.json and restarting WUD. Then after some time it happens again. I never saw this problem on previous versions, but I can't be precise if it's only on latest version or previous versions.

**Comment by fmartinou** [comment]
Issues regarding bad synchronization between wud and ha through mqtt are present from a while (probably from the beginning); multiple issues have already been opened linked to that (#97, #166, #113)
I don't think the latest version made things worse because the changes are not related to that.

Nevertheless, I need to find some time to address this point once and for all.

**Comment by alexdelprete** [comment]
Thanks. If I can help debugging/testing in any way please let me know.

**Comment by dougmaitelli** [comment]
@fmartinou Could this issue be due to MQTT container restarting and wud not handling the connection exception properly an recovering / re-connecting?
I have the same issue, over time the MQTT trigger just disappears and then just rebooting wud fixes it.

**Comment by rodrigosiviero** [comment]
Sorry, but is this related to update entities have no attributes despite being sent to mqtt?

**Comment by fmartinou** [comment]
For WUD to automatically create entities in Home Assistant via MQTT Discovery, ensure both MQTT and Home Assistant discovery are enabled:

```yaml
environment:
  - WUD_TRIGGER_MQTT_MOSQUITTO_ENABLED=true
  - WUD_TRIGGER_MQTT_MOSQUITTO_HASS_DISCOVERY=true
  - WUD_TRIGGER_MQTT_MOSQUITTO_HASS_PREFIX=homeassistant
```

Also ensure that Home Assistant's MQTT integration is configured to listen to the default discovery prefix (homeassistant).

Closing as this is standard Home Assistant MQTT integration setup.
